src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -78,6 +78,7 @@ public Context CreateContext(IFeatureCollection contextFeatures)
             }
 
             _diagnostics.BeginRequest(httpContext, hostContext);
+            httpContext.Features.Set<IActivityFeature>(new ActivityFeature());
```

Comment:
hostContext has the base activity. Why not set it onto the activity feature when it is created?